sql/debug_sync.cc

```diff
@@ -939,8 +939,8 @@ static void debug_sync_remove_action(st_debug_sync_control *ds_control,
       produced by the shift. Again do not use an assignment operator to
       avoid string allocation/copy.
     */
-    memmove(ds_control->ds_action + ds_control->ds_active, save_action,
-            sizeof(st_debug_sync_action));
+    memmove(static_cast<void*>(ds_control->ds_action + ds_control->ds_active),
```

Comment:
This does not fix the root cause of the warning. If the actual fix is too complex for us at this point of 5.5 lifecycle, just patch the build for -Wno-class-memaccess instead.